weather_providers/smhi.py

```diff
@@ -0,0 +1,138 @@
+        description_dict = {
+                        1: "Clear sky",
+                        2: "Nearly clear sky",
+                        3: "Variable cloudiness",
+                        4: "Halfclear sky",
+                        5: "Cloudy sky",
+                        6: "Overcast",
+                        7: "Fog",
+                        8: "Light rain showers",
+                        9: "Moderate rain showers",
+                        10: "Heavy rain showers",
+                        11: "Thunderstorm",
+                        12: "Light sleet showers",
+                        13: "Moderate sleet showers",
+                        14: "Heavy sleet showers",
+                        15: "Light snow showers",
+                        16: "Moderate snow showers",
+                        17: "Heavy snow showers",
+                        18: "Light rain",
+                        19: "Moderate rain",
+                        20: "Heavy rain",
+                        21: "Thunder",
+                        22: "Light sleet",
+                        23: "Moderate sleet",
+                        24: "Heavy sleet",
+                        25: "Light snowfall",
+                        26: "Moderate snowfall",
+                        27: "Heavy snowfall",
+                    }
+        description = description_dict[weathercode]
+
+        logging.debug(
+            "get_description_by_weathercode({}) - {}"
+            .format(weathercode, description))
+
+        return description.title()
+
+    # Get weather from SMHI API
+    # https://opendata.smhi.se/apidocs/metfcst/get-forecast.html#get-point-forecast
+    # The API response is a complete forecast approximately 10 days ahead of the latest current forecast.
+    # All times in the answer given in UTC.
+    # Precipitation parameters have a distribution in time (a time interval) until the valid time for current data.
+    # The interval starts at the time step before. At the beginning of the forecast, the interval is one hour.
+    # Later in the forecast, the time interval increases (eg 3, 6 and 12 h). Unit remains mm / h.
+    # So, current hour is index 0, next hour is index 1
+    # The API accepts 6 decimals in the lon/lat. With more decimals, it returns 404.
+
+    def get_weather(self):
+
+        url = ("https://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/{}/lat/{}/data.json"
+               .format(self.location_long, self.location_lat))
+
+        headers = {"User-Agent": self.smhi_self_id}
+
+        response_data = self.get_response_json(url, headers=headers)
+        logging.debug(response_data)
+        weather_data = response_data["timeSeries"]
+        logging.debug("get_weather() - {}".format(weather_data))
+
+        # Get the weather code of the first item.
+        for data in weather_data[0]["parameters"]:
+            if data["name"] == "Wsymb2":
+                weather_code = data["values"][0]
```

Comment:
In my testing the weather symbol was sometimes appearing in a different location so I decided just keep the looping bit to find the Wsymb2.